js/adapt-assessmentArticleModel.js

```diff
@@ -132,12 +132,52 @@ define([
                 }
             }
 
+            var questionBanksEnabled = this.get('_assessment')._banks._isEnabled;
+            var questionBanksLoaded = this._questionBanks;
+
+            var randomisedEnabled = this.get('_assessment')._randomisation._isEnabled;
+
             if (!quizModels) {
-                // leave the order as before, completed or not
-                quizModels = this.getChildren().models;
+
+              if ((questionBanksEnabled && !questionBanksLoaded) || randomisedEnabled) {
```

Comment:
on return from the menu quizModels is undefined but questionBanksLoaded is populated. So quizModels gets populated with all questions on line 153. 

Either quizModels should be pre-populated with this.getChildren().models (e.g. changing line 114 to
var quizModels = (this.getChildren().models.length > 0) ? this.getChildren().models : null; ) or there should be a condition for  "if ((questionBanksEnabled && questionBanksLoaded) || randomisedEnabled) {" that populates quizModels from the assessment model.